pandas/tests/tseries/test_frequencies.py

```diff
@@ -169,6 +169,15 @@ def test_to_offset_leading_zero(self):
         result = frequencies.to_offset(freqstr)
         assert (result.n == -194)
 
+    def test_to_offset_leading_plus(self):
```

Comment:
Could you add a couple invalid freqstrs, like `'+-1d'`, `'-+1d'`, `'+1'`, `'+d'` and ensure that they all raise.

At least I think they should all raise? `d` is apparently valid, but `-d` isn't.